pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -223,31 +225,42 @@ func (pl *PodTopologySpread) calPreFilterState(pod *v1.Pod) (*preFilterState, er
 		TpPairToMatchNum:     make(map[topologyPair]*int32, sizeHeuristic(len(allNodes), constraints)),
 	}
 	requiredSchedulingTerm := nodeaffinity.GetRequiredNodeAffinity(pod)
+
+	var availableNodes []*framework.NodeInfo
```

Comment:
what's the performance impact of building a filtered list here vs ignoring nodes where `tpCount == nil`

Reply:
`tpCount == nil` can't fully replace this filtered list, e.g. 
|  node   | node1  | node2  |
|  ----  | ----  | ----  |
| nodeLabel  | zoneA | zoneA |
| hasUnschedulableTaint  | yes | no |

both `node1` and `node2` has the `TpPairToMatchNum` key of `{zone: zoneA}`, so when calculating the podMatchNum, both pods located in these two nodes will be taken into consideration.